### ISSUE TITLE
unetbootin doesn't run under Wayland

### ISSUE THREAD

**Issue by Pointedstick** [unetbootin/unetbootin#94]
```
$ sudo unetbootin
No protocol specified
unetbootin: cannot connect to X server :0
```

I'm using Fedora 25 with Wayland, so there is no X server. If I log into an X session instead of Wayland, unetbootin works.

**Comment by techtonik** [comment]
On Fedora 25 the magic workaround:

1. under user run
```
$ xhost local:root
non-network local connections being added to access control list
```

2. then with `sudo`
```
$ sudo QT_X11_NO_MITSHM=1 /usr/bin/unetbootin 
```

I can not explain why it is necessary. =/

**Comment by dnylpz** [comment]
i do, x server is the program that manages access to the graphic devices drivers, xhosts allows users to connect to those drivers.

since sudo doesn't have an active session nor it's allow to access the server, unetbootin cannot create an instance of it's GUI

once you run that you give sudo access to the device drivers and hence allow unetbootin (since it's running as sudo) to create a nice looking window :)

**Comment by Pointedstick** [comment]
The workaround is not sustainable, though. unetbootin needs to use Polkit rather than demanding to be run as root.

**Comment by DonyorM** [comment]
Even with Polkit, Wayland won't let a GUI run as root (not without using "legacy" options at least). I've had trouble finding the suggested way to get around this.

**Comment by JonnyTech** [comment]
This works for me: `$ xhost +local`

**Comment by Pointedstick** [comment]
@DonyorM the point is, it would need to be rewritten to run as a normal user, and use polkit to request elevated privileges when necessary.

**Comment by DonyorM** [comment]
Yeah, that's what I was getting at. I wish there was more documentation on how one does that though. Polkit docs are a bit messy.

**Comment by JtLively** [comment]
I had to add the user to the **sudoers** file.

**Comment by johnnyasantoss** [comment]
What worked for me
```bash
beesu xhost `hostname -s`
beesu QT_X11_NO_MITSHM=1 unetbootin
```

**Comment by cialu** [comment]
No working workaround for me on Fedora 26, anyway get:
```
No protocol specified
unetbootin: cannot connect to X server :0
```

**Comment by bicyu** [comment]
Techtonik's answer worked for me on Ubuntu 17.10.
My problem was the app would open, ask for p/w then close.

**Comment by kaetschap** [comment]
same happened to me on Fedora 27, but @techtonik 's answer worked, thanks!

**Comment by bicyu** [comment]
Maybe this won't work for everyone but I tired of having to sudo each time I used Unetbootin. By a fluke chance I found it worked when I logged into Ubuntu 17.10 using the old XOrg option, **NOT** Ubuntu.

**Comment by rubo77** [comment]
You can add an alias to your bashrc:

     alias unetbootin="xhost local:root && sudo QT_X11_NO_MITSHM=1 unetbootin"

**Comment by bicyu** [comment]
Nice work rubo77, hope it helps some people. 
Sadly on Ubuntu 17.1 I still get:

> Failed to run /usr/bin/unetbootin 'rootcheck=no' as user root. Unable to copy the user's Xauthorization file. 
Not such an issue now tho, as I can access it if I log in via Xorg

**Comment by bicyu** [comment]
Does anyone know how to shortcut to desktop on Ubu 17-10 please?
I usually work with at least 8 screens open and it is a real pain in the bottom having to shrink all of them when i need the desk top - which is often.

Thx

**Comment by aminvakil** [comment]
@bicyu I guess you need WIN Key + d, if it doesn't work try Ctrl + Win Key + d.

**Comment by bicyu** [comment]
@aminvakil Cheers for that mate, both work. 
After a lot of trail and error i finally found ct+alt+d does the same job- at least for me. However, I prefer your 2 key win+d approach.
Oh, works in 18.04 too.
BTW, for anyone who has / had issues with Wayland in 17.10, the new 18.04 has reverted to Xorg so most problems seem to have been solved. Looking at YOU Unetbootin~!
My 18.4 does seem a wee bit slower than 17.10 though, and I have the "light" version
